src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCommandCommand.java

```diff
@@ -177,84 +127,56 @@ public static Command matchCommandByName(ScriptEntry scriptEntry, ElementTag nam
-    public void execute(ScriptEntry scriptEntry) {
-        DiscordBotTag bot = scriptEntry.requiredArgForPrefix("id", DiscordBotTag.class);
-        ElementTag commandInstruction = scriptEntry.getElement("instruction");
-        DiscordGroupTag group = scriptEntry.getObjectTag("group");
-        ElementTag name = scriptEntry.getElement("name");
-        ElementTag description = scriptEntry.getElement("description");
-        MapTag options = scriptEntry.getObjectTag("options");
-        ElementTag enabled = scriptEntry.getElement("enabled");
-        ListTag enableFor = scriptEntry.getObjectTag("enable_for");
-        ListTag disableFor = scriptEntry.getObjectTag("disable_for");
-        ElementTag type = scriptEntry.argForPrefixAsElement("type", "slash");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), bot, commandInstruction, group, name, description, options, enabled, enableFor, disableFor, type);
-        }
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgPrefixed @ArgName("id") DiscordBotTag bot,
+                                   @ArgName("instruction") DiscordCommandInstruction instruction,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("group") DiscordGroupTag group,
+                                   @ArgPrefixed @ArgName("name") String name,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("type") Command.Type type,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("description") String description,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("options") MapTag options,
+                                   // Past-deprecated arguments
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("enabled") ElementTag enabled,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("enable_for") ListTag enableFor,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("disable_for") ListTag disableFor) {
         if (group != null && group.bot == null) {
             group.bot = bot.bot;
         }
         JDA client = bot.getConnection().client;
         Bukkit.getScheduler().runTaskAsynchronously(DenizenDiscordBot.instance, () -> {
             try {
-                if (commandInstruction == null) {
+                Command.Type commandType = type != null ? type : Command.Type.SLASH;
+                if (instruction == null) {
                     Debug.echoError(scriptEntry, "Must have a command instruction!");
                     scriptEntry.setFinished(true);
                     return;
                 }
+                if (enabled != null || enableFor != null || disableFor != null || instruction == DiscordCommandInstruction.PERMS) {
+                    DenizenDiscordBot.oldCommandPermissions.warn(scriptEntry);
+                }
                 else if (name == null) {
```

Comment:
this `else` keyword is no longer valid due to the interjection of a separate `if` in the middle